scripts/build_a_package.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script builds wheels for a single package when trigged from a push to
+# a tag as part of a GitHub workflow (See .github/publish-a-package.yml). The
+# wheel is then published to PyPI.
+
+set -ev
+
+if [[ -z $GITHUB_REF ]]; then
+  echo 'Failed to run script, missing workflow env variable GITHUB_REF'
+  exit -1
+fi
+
+# Follows PEP 440 versioning conventions: https://www.python.org/dev/peps/pep-0440/#pre-releases
+# Only allow publishing packages that are 1.0+ using this method
+if ! [[ $GITHUB_REF =~ ^refs/tags/(opentelemetry-.*)==([1-9]([0-9]+)?\.[0-9]+((a|b|rc)[0-9]+)?(\.(post)?[0-9]+)?)$ ]]; then
+  echo 'Failed to parse package name and package version from tag.'
+  exit -1
+fi
+
+# From the capture groups in the regex above
+PKG_NAME=${BASH_REMATCH[1]}
+PKG_VERSION=${BASH_REMATCH[2]}
+
+# Get the latest versions of packaging tools
+python3 -m pip install --upgrade pip setuptools wheel
+
+# If testing locally, change `BASEDIR` to your own repo directory
+BASEDIR=$(dirname $(readlink -f $(dirname $0)))
```

Comment:
Why would you need to change it when testing locally, this should work fine either way?

nit I've found this oneliner to be a bit simpler in other scripts: https://github.com/open-telemetry/opentelemetry-python/blob/c49e6fd6417282934b2ec4320b62b0e3adbe082d/scripts/proto_codegen.sh#L20-L21

Reply:
Sorry I probably should have been more specific, here's the error I get when running with `/bin/sh`:

```sh
$ sh
sh-3.2$ BASEDIR=$(dirname $(readlink -f $(dirname $0)))
readlink: illegal option -- f
usage: readlink [-n] [file ...]
usage: dirname path
```

That line you provided works great though! I'll be using it, thanks 🙂